### PR TITLE
Extend build with transport

### LINKED ISSUE #7: Support for transport
Issue:
- support for transport in DFTB+ conda-forge package (`WITH_TRANSPORT` option)
- requires [`libnegf`](https://github.com/conda-forge/libnegf-feedstock) dependency:
```yaml
requirements:
  host:
    - libnegf
    - libnegf * mpi_{{ mpi }}_*  # [ mpi != "nompi ]
  run:
    - pin_compatible('libnegf', max_pin('x.x'))  # [ mpi == "nompi" ]
    - pin_compatible('libnegf', max_pin('x.x')) mpi_{{ mpi }}_*  # [ mpi != "nompi" ]
```

### PR DESCRIPTION
**Pull request by aradi**
Also use the customized source archive instead of GitHubs
automatically generated one to ensure that the RELEASE file is
contained.

Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [X] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by aradi** [comment]
@conda-forge-admin please rerender

**Comment by aradi** [comment]
@conda-forge-admin please rerender

**Comment by awvwgk** [comment]
CMake's detection for MPI is currently broken in mpich 3.4.x (https://github.com/conda-forge/mpich-feedstock/issues/56).

Pinning mpich to not 3.4.x in host seems to work for now, but is quite ugly:

```yaml
  host:
    - {{ mpi }}  # [mpi != "nompi"]
    - mpich !=3.4  # [mpi == "mpich"]
```

I would recommend to wait for upstream to fix this issue instead of messing with the host dependencies.

**Comment by aradi** [comment]
@awvwgk Do you have by any chance clue, why the OSX with MPICH fails? I can not see, why any of my changes should have caused this failure... (I guess, OSX and me are not really compatible for some reasons :wink: )

**Comment by aradi** [comment]
Makes sense, the PR is not too urgent.

**Comment by awvwgk** [comment]
@conda-forge-admin, please restart CI

**Comment by awvwgk** [comment]
@aradi what is the status on this patch? The CI is passing with the MPICH fix so we could move this forward now.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -42,6 +42,7 @@ requirements:
     - mpifx * mpi_{{ mpi }}_*  # [mpi != "nompi"]
     - scalapack  # [mpi != "nompi"]
     - scalapackfx * mpi_{{ mpi }}_*  # [mpi != "nompi"]
+    - libnegf * {{ mpi_prefix }}_*
```
Comment (awvwgk) [inline recipe/meta.yaml:45]: The correct `libnegf` version is picked automatically for non-MPI builds, constraints only apply for MPI builds:
```suggestion
    - libnegf
    - libnegf * mpi_{{ mpi }}_*  # [mpi != "nompi"]
```

**Review by awvwgk — CHANGES_REQUESTED** [review]
Also `libnegf` should be pinned in the run dependencies correctly.

**Comment by awvwgk** [comment]
@aradi What is the status on this PR?

**Review by awvwgk — APPROVED** [review]

**Pull request merged**